moto/cloudformation/models.py

```diff
@@ -712,6 +712,11 @@ def delete_stack(self, name_or_stack_id):
         if name_or_stack_id in self.stacks:
             # Delete by stack id
             stack = self.stacks.pop(name_or_stack_id, None)
+            template = stack.template_dict
```

Comment:
Resources should already be deleted as part of the `stack.delete()` call: https://github.com/spulec/moto/blob/master/moto/cloudformation/models.py#L363 , so I don't think these lines are necessary